data/src/ctcp.rs

```diff
@@ -15,10 +15,13 @@ pub fn parse_query(text: &str) -> Option<Query> {
         .strip_prefix('\u{1}')?;
 
     if let Some((command, params)) = query.split_once(char::is_whitespace) {
-        Some(Query { command, params })
+        Some(Query {
+            command: command.to_uppercase(),
+            params,
+        })
     } else {
         Some(Query {
-            command: query,
+            command: query.to_uppercase(),
```

Comment:
We are relying on an internal implementation detail from another module, to know we should match on this "uppercase".

Instead, let's define a Command enum in this module and match on this uppercase guarantee _internally_ to produce the strongly typed enum. `client` module can then rely on the strongly typed Command guarantee, making this impossible to misuse.